packages/valory/skills/inbox_abci/rounds.py

```diff
@@ -79,6 +79,9 @@ class WaitRound(CollectSameUntilThresholdRound):
     def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
         """Process the end of the block."""
         if self.threshold_reached:
+            # TODO: suboptimal to use json encoding. You then don't get the framework
+            # mechanisms that ensure message integrity for free. In a multi-agent
+            # system this could fail in many ways (not json, not right fields, etc etc)
             payload = json.loads(
```

Comment:
My biggest issue. @angrybayblade and @0xArdi curious to hear your thoughts

Reply:
Unless there have been changes that I am unaware of, you can't pass around objects, including dicts. It is due to the way payloads are encoded/decoded when verifying signatures.

Reply:
You can have arbitrarily many message fields though. Which ensures that messages with identical field values are considered identical.